docker, runtime: only build clang and llc targets

tklauser opened the pull request:
Specify only the targets we need from the llvm build, namely `clang` and
`llc`. This reduces the amount of objects to build by ~300 and thus
also reduces build time a bit.

(Edit from @borkmann: added the test update commit which fixes below issue)

Fixes: #10882

tklauser committed:
docker, runtime: only build clang and llc targets

Specify only the targets we need from the llvm build, namely `clang` and
`llc`. This reduces the amount of objects to build by ~300 and thus
also reduces build time a bit.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

aanm commented on contrib/packaging/docker/Dockerfile.runtime:
```diff
@@ -58,7 +58,7 @@ cd llvm/ && \
 cd llvm/build && \
 cmake .. -G "Ninja" -DLLVM_TARGETS_TO_BUILD="BPF" -DLLVM_ENABLE_PROJECTS="clang" -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_RUNTIME=OFF && \
-ninja && \
+ninja clang llc && \
```
Needs a new image tag for the runtime image. Do you want to do it part of this PR?

tklauser replied:
Thanks, forgot about this. Will trigger a new build and tag the image.

borkmann committed:
tests, services: update pod loopback bpf nodeport tests

After updating the cilium-runtime image we also get a newer bpftool
build which generates a bpf_features.h header that includes the netns
cookie and therefore on bpf-next kernel we must fail this particular
test.

Similarly, doFragmentedRequest() from pod loopback does not make sense
either anymore. Given we performed the earlier testCurlRequestFail()
for a service, there is no point in retesting a fragmented request
again for this address.

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
